### PR TITLE
Kad inspector fixes

### PR DESCRIPTION
**Pull request by bryanchriswhite**

### CODE REVIEW

File: pkg/kademlia/kademlia.go [hunk 1]
```diff
@@ -329,6 +329,9 @@ func Restrict(r pb.Restriction, n []*pb.Node) []*pb.Node {
 
 func meetsRestrictions(rs []pb.Restriction, n pb.Node) bool {
 	for _, r := range rs {
+		if proto.Equal(&r, &pb.Restriction{}) {
+			return true
```
Comment (egonelbre) [inline pkg/kademlia/kademlia.go]: Why should this return true?

Comment (dylanlott) [reply pkg/kademlia/kademlia.go]: If an empty restriction was passed through, it caused no nodes to match. We want the opposite behavior.

Comment (egonelbre) [reply pkg/kademlia/kademlia.go]: I mean, why not pass no restrictions for that case rather than passing in the empty restriction?

Comment (dylanlott) [reply pkg/kademlia/kademlia.go]: I think it's better to be explicit than implicit in this case. It's not like it's less performant or less clear.

File: pkg/inspector/server.go [hunk 2]
```diff
@@ -45,7 +45,10 @@ func (srv *Server) CountNodes(ctx context.Context, req *pb.CountNodesRequest) (*
 	if err != nil {
 		return nil, err
 	}
-	kadNodes := srv.dht.Seen()
+	kadNodes, err := srv.dht.GetNodes(ctx, srv.identity.ID, 0, pb.Restriction{})
```
[the remark this answers is not in the document]
Comment (bryanchriswhite) [reply pkg/inspector/server.go]: That parameter is not a string anymore. I don't think an empty node ID returns any results.

File: pkg/kademlia/kademlia.go (hunk 1 above)
Comment (egonelbre) [reply pkg/kademlia/kademlia.go]: To me it's inconsistent. The previous constraint checking was to `&` together all the restrictions together, this is clear. However now it's `&` except when you have case Z.

Also compare:
```
srv.dht.GetNodes(ctx, srv.identity.ID, 5, pb.Restriction{...}) // with a restriction
srv.dht.GetNodes(ctx, srv.identity.ID, 5) // without restrictions
```

The latter is shorter and quite clear that there isn't a restriction in place.

**Pull request merged**